pyuvdata/uvbeam.py

```diff
@@ -0,0 +1,1023 @@
+                '(Naxes_vec, 2, Npixels), otherwise it is (Naxes_vec, 2, Naxes1, Naxes2)')
+        self._basis_vector_array = uvp.UVParameter('basis_vector_array',
+                                                   description=desc, required=False,
+                                                   expected_type=np.float,
+                                                   form=('Naxes_vec', 2, 'Naxes2', 'Naxes1'),
+                                                   acceptable_range=(0, 1),
+                                                   tols=1e-3)
+
+        self._Nfeeds = uvp.UVParameter('Nfeeds', description='Number of feeds. '
+                                       'Not required if beam_type is "power".',
+                                       expected_type=int, acceptable_vals=[1, 2],
+                                       required=False)
+
+        desc = ('Array of feed orientations. shape (Nfeeds). '
+                'options are: N/E or x/y or R/L. Not required if beam_type is "power".')
+        self._feed_array = uvp.UVParameter('feed_array', description=desc, required=False,
+                                           expected_type=str, form=('Nfeeds',),
+                                           acceptable_vals=['N', 'E', 'x', 'y',
+                                                            'R', 'L'])
+
+        self._Npols = uvp.UVParameter('Npols', description='Number of polarizations. '
+                                      'Only required if beam_type is "power".',
+                                      expected_type=int, required=False)
+
+        desc = ('Array of polarization integers, shape (Npols). '
+                'AIPS Memo 117 says: stokes 1:4 (I,Q,U,V);  '
+                'circular -1:-4 (RR,LL,RL,LR); linear -5:-8 (XX,YY,XY,YX). '
+                'Only required if beam_type is "power".')
+        self._polarization_array = uvp.UVParameter('polarization_array',
+                                                   description=desc, required=False,
+                                                   expected_type=int, form=('Npols',),
+                                                   acceptable_vals=list(np.arange(-8, 0)) + list(np.arange(1, 5)))
+
+        desc = 'Array of frequencies, shape (Nspws, Nfreqs), units Hz'
+        self._freq_array = uvp.UVParameter('freq_array', description=desc,
+                                           form=('Nspws', 'Nfreqs'),
+                                           expected_type=np.float,
+                                           tols=1e-3)  # mHz
+
+        self._spw_array = uvp.UVParameter('spw_array',
+                                          description='Array of spectral window '
+                                          'Numbers, shape (Nspws)', form=('Nspws',),
+                                          expected_type=int)
+
+        desc = 'Normalization standard of data_array, options are: "peak", "solid_angle"'
```

Comment:
I'm embarrassed to say that I don't remember exactly what it means, it was something Dave, James and I came up with. I think it might be related to the isotropic radiator, I've pinged them to get help remembering.